esphome/components/lvgl/animimg.py

```diff
@@ -0,0 +1,117 @@
+from esphome import automation
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.const import CONF_DURATION, CONF_ID
+
+from ...cpp_generator import MockObj
+from .automation import action_to_code
+from .defines import CONF_AUTO_START, CONF_MAIN, CONF_REPEAT_COUNT, CONF_SRC
+from .helpers import lvgl_components_required
+from .img import CONF_IMAGE
+from .label import CONF_LABEL
+from .lv_validation import lv_image, lv_milliseconds
+from .lvcode import lv, lv_expr
+from .types import LvType, ObjUpdateAction, void_ptr
+from .widget import Widget, WidgetType, get_widgets
+
+CONF_ANIMIMG = "animimg"
+CONF_SRC_LIST_ID = "src_list_id"
+
+
+def lv_repeat_count(value):
+    if isinstance(value, str) and value.lower() in ("forever", "infinite"):
+        value = 0xFFFF
+    return cv.int_range(min=0, max=0xFFFF)(value)
+
+
+ANIMIMG_BASE_SCHEMA = cv.Schema(
+    {
+        cv.Optional(CONF_REPEAT_COUNT, default="forever"): lv_repeat_count,
+        cv.Optional(CONF_AUTO_START, default=True): cv.boolean,
+    }
+)
+ANIMIMG_SCHEMA = ANIMIMG_BASE_SCHEMA.extend(
+    {
+        cv.Required(CONF_DURATION): lv_milliseconds,
+        cv.Required(CONF_SRC): cv.ensure_list(lv_image),
+        cv.GenerateID(CONF_SRC_LIST_ID): cv.declare_id(void_ptr),
+    }
+)
+
+ANIMIMG_MODIFY_SCHEMA = ANIMIMG_BASE_SCHEMA.extend(
+    {
+        cv.Optional(CONF_DURATION): lv_milliseconds,
+    }
+)
+
+lv_animimg_t = LvType("lv_animimg_t")
+
+
+class AnimimgType(WidgetType):
+    def __init__(self):
+        super().__init__(
+            CONF_ANIMIMG,
+            lv_animimg_t,
+            (CONF_MAIN,),
+            ANIMIMG_SCHEMA,
+            ANIMIMG_MODIFY_SCHEMA,
+        )
+
+    async def to_code(self, w: Widget, config):
+        lvgl_components_required.add(CONF_IMAGE)
+        lvgl_components_required.add(CONF_ANIMIMG)
+        if CONF_SRC in config:
+            for x in config[CONF_SRC]:
+                await cg.get_variable(x)
+            srcs = [lv_expr.img_from(MockObj(x)) for x in config[CONF_SRC]]
+            src_id = cg.static_const_array(config[CONF_SRC_LIST_ID], srcs)
+            count = len(config[CONF_SRC])
+            lv.animimg_set_src(w.obj, src_id, count)
```

Comment:
`CONF_SRC` is Required so it will always be there
```suggestion
        for x in config[CONF_SRC]:
            await cg.get_variable(x)
        srcs = [lv_expr.img_from(MockObj(x)) for x in config[CONF_SRC]]
        src_id = cg.static_const_array(config[CONF_SRC_LIST_ID], srcs)
        count = len(config[CONF_SRC])
        lv.animimg_set_src(w.obj, src_id, count)
```